Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -169,41 +169,86 @@
+##### With Source Code
 
--- TODO [Add content on static analysis of "Testing JavaScript Execution in WebViews" with source code] --
+The WKWebView object allows for JavaScript execution by default. That may raise a serious risk of running arbitrary code on user's device via WebView object. If your application does not require executing JavaScript (just display a web page), you should definitely disable it. You can do it using preferences of an object WKPreferences<sup>[3]</sup>, like in the following example:
+
+```
+#import "ViewController.h"
+#import <WebKit/WebKit.h>
+@interface ViewController ()<WKNavigationDelegate,WKUIDelegate>
+@property(strong,nonatomic) WKWebView *webView;
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
+    
+    NSURL *url = [NSURL URLWithString:@"http://www.example.com/"];
+    NSURLRequest *request = [NSURLRequest requestWithURL:url];
+    WKPreferences *pref = [[WKPreferences alloc] init];
+    
+    //Disable javascript execution:
+    [pref setJavaScriptEnabled:NO];
+    [pref setJavaScriptCanOpenWindowsAutomatically:NO];
+    
+    WKWebViewConfiguration *conf = [[WKWebViewConfiguration alloc] init];
+    [conf setPreferences:pref];
+    _webView = [[WKWebView alloc]initWithFrame:CGRectMake(self.view.frame.origin.x,85, self.view.frame.size.width, self.view.frame.size.height-85) configuration:conf] ;
+    [_webView loadRequest:request];
+    [self.view addSubview:_webView];
+    
+}
+
+```
+
+If there is no explicitly disabled JavaScript execution via WKPreferences object, then it means it is enabled.
 
 
 #### Dynamic Analysis
 
--- TODO [Describe how to test for this issue "Testing JavaScript Execution in WebViews" by running and interacting with the app. This can include everything from simply monitoring network traffic or aspects of the app’s behavior to code injection, debugging, instrumentation, etc.] --
+A Dynamic Analysis depends on different surrounding conditions, as there are different possibilities to inject JavaScript into a WebView of an application:
+
+* Stored Cross-Site Scripting (XSS) vulnerability in an endpoint, where the exploit will be sent to the WebView of the Mobile App when navigating to the vulnerable function.
+* Man-in-the-middle (MITM) position by an attacker where he is able to tamper the response by injecting JavaScript.
 
 #### Remediation
 
--- TODO [Describe the best practices that developers should follow to prevent this issue "Testing JavaScript Execution in WebViews".] --
+The UIWebView should be avoided and WKWebView used instead. JavaScript is enabled by default in a WKWebView and should be disabled if not needed. This reduces the attack surface and potential threats to the application. If JavaScript is needed it should be ensured:
+
+In order to address these attack vectors, the outcome of the following checks should be verified:
```

Comment:
I would remove this sentence as already the sentence before "If JavaScript is needed it should be ensured:" is referring to the bullet points below.